lib/multiplexer/multiplexer_test.go

```diff
@@ -281,6 +281,55 @@ func (s *MuxSuite) TestTimeout(c *check.C) {
 	c.Assert(err, check.NotNil)
 }
 
+// TestMultiplexerHealthcheck makes sure that the multiplexer replies with a
+// healthcheck response to an HTTP GET request
+func (s *MuxSuite) TestMultiplexerHealthcheck(c *check.C) {
+	ports, err := utils.GetFreeTCPPorts(1)
+	c.Assert(err, check.IsNil)
+
+	listener, err := net.Listen("tcp", net.JoinHostPort("127.0.0.1", ports[0]))
```

Comment:
I think binding to `127.0.0.1:0` and then reading the port from `listener.Addr()` is safer - here's still a chance of a race between `GetFreeTCPPorts` and use of the port.